compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java

```diff
@@ -139,19 +133,23 @@ PackageContext packageContext() {
         return rootPackageContext;
     }
 
-    private void compile() {
-        synchronized (this.compilerContext) {
-            List<Diagnostic> diagnostics = new ArrayList<>();
-            for (ModuleContext moduleContext : packageResolution.topologicallySortedModuleList()) {
-                moduleContext.compile(compilerContext);
-                moduleContext.diagnostics()
-                        .forEach(diagnostic -> diagnostics
-                                .add(new PackageDiagnostic(diagnostic, moduleContext.moduleName())));
+    private void compileIfRequired() {
+        if (!compiled) {
```

Comment:
Can we reverse this if condition?

Reply:
inverted the conditions with https://github.com/ballerina-platform/ballerina-lang/pull/28617/commits/359ad05f162197aeb177afbae62949e7963aa024